finesse/gui/path_widget.py

```diff
@@ -0,0 +1,141 @@
+"""Provides a widget for choosing the path to a measure script."""
+from abc import abstractmethod
+from pathlib import Path
+from typing import Any, Optional
+
+from PySide6.QtWidgets import (
+    QFileDialog,
+    QHBoxLayout,
+    QLineEdit,
+    QPushButton,
+    QSizePolicy,
+    QWidget,
+)
+
+
+class PathWidget(QWidget):
+    """A widget containing a text box with a file path and a browse button."""
+
+    def __init__(self, initial_file_path: Optional[Path] = None) -> None:
+        """Create a new PathWidget.
+
+        Args:
+            initial_file_path: The initial file path to display
+        """
+        super().__init__()
+
+        self.line_edit = QLineEdit()
+        """Indicates the current selected path."""
+
+        self.line_edit.setReadOnly(True)
```

Comment:
Perhaps it might be useful to let the user enter a path too? I know I get frustrated navigating through file dialogs. Though I realise one should minimise human input as much as possible!

Reply:
True!